ui/components/component-library/button-base/button-base.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import classnames from 'classnames';
+import { IconName, Icon, IconSize, Text } from '..';
+import {
+  AlignItems,
+  Display,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+  BorderRadius,
+  BackgroundColor,
+  IconColor,
+} from '../../../helpers/constants/design-system';
+import type { PolymorphicRef } from '../box';
+import type { TextProps } from '../text';
+import {
+  ButtonBaseProps,
+  ButtonBaseSize,
+  ButtonBaseComponent,
+} from './button-base.types';
+
+export const ButtonBase: ButtonBaseComponent = React.forwardRef(
+  <C extends React.ElementType = 'button' | 'a'>(
+    {
+      as,
+      block,
+      children,
+      className = '',
+      href,
+      ellipsis = false,
+      externalLink,
+      size = ButtonBaseSize.Md,
+      startIconName,
+      startIconProps,
+      endIconName,
+      endIconProps,
+      loading,
+      disabled,
+      iconLoadingProps,
+      textProps,
+      color = TextColor.textDefault,
+      iconColor = IconColor.iconDefault,
+      ...props
+    }: ButtonBaseProps<C>,
+    ref?: PolymorphicRef<C>,
+  ) => {
+    const Tag = href ? 'a' : as || 'button';
```

Comment:
Small nit: Because Tag is just being used as a string and not rendered as a component e.g. `<Tag>` might we use lower case here?
```suggestion
    const tag = href ? 'a' : as || 'button';
```